index.js

```diff
@@ -183,13 +183,11 @@ function _decodeLogs(logs) {
           param.type === "uint8" ||
           param.type === "int"
         ) {
-          // ensure to remove leading 0x for hex numbers
           if (typeof decodedP.value === "string" && decodedP.value.startsWith("0x")) {
-            decodedP.value = new BN(decodedP.value.slice(2), 16).toString(10);
+            decodedP.value = new BN(decodedP.value, 16).toString(10);
           } else {
             decodedP.value = new BN(decodedP.value).toString(10);
```

Comment:
2 branches are now identical because `web3-utils/BN` ignores the second parameter (it already handled `0x` and passed `16` as second param).


```js
if (typeof decodedP.value === "string") {
  decodedP.value = new BN(decodedP.value).toString(10);
}
```